File: app/src/main/java/com/baozi/mvp/view/UIView.java
package com.baozi.mvp.view;

import android.content.res.Resources;
import android.support.annotation.IdRes;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.view.Window;

/**
 * Created by baozi on 2017/2/20.
 * 用户页面,操作页面，对应Activity,frgament...
 */

public interface UIView extends BaseView {
    /**
     * res资源获取
     *
     * @return
     */
    Resources getResources();

    /**
     * 根据id获得控件
     * 需要调用在initView()之后,否则会出现NullPointerException
     *
     * @param viewId
     * @param <V>
     * @return
     */

    <V extends View> V findView(@IdRes int viewId);

    /**
     * 回退
     */

    void onBackPressed();

    /**
     * 获取Activity
     *
     * @return
     */
    AppCompatActivity getAppcompatActivity();

    /**
     * 主要视图View
     *
     * @return
     */
    View getContentView();

    /**
     * 处理异常
     *
     * @param throwable
     */
    void onNewThrowable(Throwable throwable);

    /**
     * Frgament跳转.
     *
     * @param fragment
     */
    void startFragment(Fragment fragment);

    /**
     * Frgament跳转.
     *
     * @param fragment
     */
    void startFragment(Fragment fragment, String tag);

    void startFragment(Fragment fragment, String tag, boolean isAdd);

    void startFragment(Fragment fragment, String tag, int enterAnim, int exitAnim);

    void startFragment(Fragment fragment, String tag, int enter, int popExit, boolean isAddBack);

    void startFragment(Fragment fragment, String tag, int enterAnim, int exitAnim, int popEnter, int popExit, boolean isAddBack);

    Window getWindow();

    ActionBar getSupportActionBar();

    void setSupportActionBar(@Nullable Toolbar toolbar);

    void finishActivity();
}

File: lib-mvpbase-kotlin/src/main/java/com/linfeng/mvp/view/UIView.kt
package com.linfeng.mvp.view

import android.content.res.Resources
import android.support.v4.app.Fragment
import android.support.v7.app.ActionBar
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.View
import android.view.Window

/**
 * Created by baozi on 2017/2/20.
 * 用户页面,操作页面，对应Activity,frgament...
 */

interface UIView : BaseView {
    /**
     * res资源获取
     *
     * @return
     */
    fun resources(): Resources

    /**
     * 获取Activity
     *
     * @return
     */
    fun appcompatActivity(): AppCompatActivity

    /**
     * 主要视图View
     *
     * @return
     */
    fun getContentView(): View

    fun window(): Window

    fun supportActionBar(): ActionBar?

    /**
     * 回退
     */

    fun onBack()

    /**
     * 处理异常
     *
     * @param throwable
     */
    fun onNewThrowable(throwable: Throwable)

    /**
     * Frgament跳转.
     *
     * @param fragment
     */
    fun startFragment(fragment: Fragment)

    /**
     * Frgament跳转.
     *
     * @param fragment
     */
    fun startFragment(fragment: Fragment, tag: String? = null, enterAnim: Int = 0, exitAnim: Int = 0, popEnter: Int = 0
                      , popExit: Int = 0, isAddBack: Boolean = true)

    fun setSupportActionBar(toolbar: Toolbar?)

    fun finishActivity()
}
